### DISCUSSION TITLE
Search query "plumbers near Chicago"

### DISCUSSION THREAD

**Discussion by mshussein** [meilisearch/product#750] [Feedback & Feature Proposal]
I've looked at documentation everywhere, but yet to find an answer.

I have a marketplace website with a list of service providers and their addresses already geocoded.

Can meilisearch ascertain that a search query "plumbers near Chicago" requires a geocoding of "Chicago" and subsequently search using the geopoint of Chicago and some arbitrary radius?

Or another one, "plumbers within 50 miles of Chicago" where the radius is specified in the query itself.

Currently, we've resorted to having 3 fields in UI:
1. search term
2. location, geocoded with Google Maps
3. radius slider

And we'd like to merge this all into 1 field, much like a modern search engine. Can meilisearch do this?

**Comment by macraig** [comment]
Hi @mshussein , unfortunately, Meilisearch doesn't support natural language querying for geolocated searches.
